### PR TITLE
fix(pd): fix the pd name of tc with across k8s opt

### PR DESCRIPTION
**Pull request by liubog2008**
### What problem does this PR solve?

Now if `spec.acrossK8s` is set but `spec.clusterDomain` is empty in TC, the upgrading process of PD will be blocked because the name of pd in status is mismatched with the return result of function `PdName`.

For now, if `spec.clusterDomain` is empty, the `PdName` will return pod name whether `spec.acrossK8s` is true or false. However, if `spec.acrossK8s` is true, the real pd name generated by start-script is the full domain name.

After this PR merged, `PdName` will also return full domain name when `spec.acrossK8s` is true

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [x] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/manager/member/utils.go [hunk 1]
```diff
@@ -140,10 +141,16 @@ func DMMasterPodName(dcName string, ordinal int32) string {
...
+func PdName(tcName string, ordinal int32, namespace string, clusterDomain string, acrossK8s bool) string {
 	if len(clusterDomain) > 0 {
 		return fmt.Sprintf("%s.%s-pd-peer.%s.svc.%s", PdPodName(tcName, ordinal), tcName, namespace, clusterDomain)
 	}
+
+	// clusterDomain is not set
+	if acrossK8s {
+		return fmt.Sprintf("%s.%s-pd-peer.%s.svc", PdPodName(tcName, ordinal), tcName, namespace)
+	}
+
```
Comment (KanShiori) [inline pkg/manager/member/utils.go:153]: how about adding a comment to reference pd name in the start script?

**Review by KanShiori — APPROVED** [review]

File: pkg/manager/member/utils.go (hunk 1 above)
Comment (liubog2008) [reply pkg/manager/member/utils.go:153]: OK

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**